internal/database/backup.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2021 - 2023, Ludvig Lundgren and the autobrr contributors.
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/autobrr/autobrr/pkg/errors"
+)
+
+var backupPrefix = "autobrr.db.backup."
+
+func (db *DB) BackupDatabase(shuttingDown bool) error {
+	if db.handler == nil {
+		return errors.New("backup: invalid database handle")
+	}
+
+	switch db.Driver {
+	case "sqlite":
+		opts := &sql.TxOptions{
+			Isolation: sql.LevelSnapshot,
+			ReadOnly:  true,
+		}
+
+		base := "./"
+		if idx := strings.LastIndexByte(db.DSN, byte('/')); idx != -1 {
+			base = db.DSN[:idx]
+		}
+
+		tx, err := db.handler.BeginTx(context.Background(), opts)
+		if err != nil {
+			return errors.Wrap(err, "backup: Transaction creation failed")
+		}
+
+		defer tx.Commit()
+		if err := databaseConsistent(tx); err != nil {
+			return err
+		}
+
+		if exists, err := backupDatabase(base, db.handler); err != nil {
+			if !exists {
+				return err
+			}
+
+			return nil
+		}
+
+		retain := 1
+		if !shuttingDown {
+			retain++
+		}
+
+		return cleanupDatabase(base, db, retain)
+	}
+
+	return errors.New("backup: not implemented for database type: %s", db.Driver)
+}
+
+func databaseConsistent(tx *sql.Tx) error {

```

Comment:
Should probably make this clear it's for SQLite only, so rename it to `databaseConsistentSQLite` might be a good idea.